lab-advance-querying-mongo-master/queries.md

```diff
@@ -0,0 +1,177 @@
+![Ironhack Logo](https://i.imgur.com/1QgrNNw.png)
+
+
+# Answers
+
+### 1. All the companies whose name match 'Babelgum'. Retrieve only their `name` field.
+
+<!-- 
+
+Filter: {name: "Babelgum"}
+Project: {name: 1}
+
+ -->
+
+### 2. All the companies that have more than 5000 employees. Limit the search to 20 companies and sort them by **number of employees**.
+
+<!-- 
+
+Filter: { number_of_employees: { $gt: 5000 } }
+Sort: {number_of_employees: 1}
+Limit: 20
+
+ -->
+
+### 3. All the companies founded between 2000 and 2005, both years included. Retrieve only the `name` and `founded_year` fields.
+
+<!-- 
+
+Filter: { $and: [ {founded_year: { $gt: 1999} }, {founded_year: { $lt: 2006} } ] }
+Project: {name: 1, founded_year: 1}
+Sort: {founded_year: 1}
+
+
+ -->
+
+### 4. All the companies that had a Valuation Amount of more than 100.000.000 and have been founded before 2010. Retrieve only the `name` and `ipo` fields.
+
+<!-- 
+
+Filter: { $and: [ {"ipo.valuation_amount": { $gt: 99999999} }, { founded_year: {$lt: 2010} }  ] }
+Project: {"ipo.valuation_amount": 1, name: 1}
+Sort: {founded_year: 1}
+
+ -->
+
+### 5. All the companies that have less than 1000 employees and have been founded before 2005. Order them by the number of employees and limit the search to 10 companies.
+
+<!-- 
+
+Filter: { $and: [ {number_of_employees: { $lt: 1000 } }, { founded_year: { $gt: 2005 } } ] }
+Sort: {number_of_employees: 1}
+Limit: 10
+
+ -->
+
+### 6. All the companies that don't include the `partners` field.
+
+<!-- 
+
+Filter: { partners: { $exists: false } }
+
+ -->
+
+### 7. All the companies that have a null type of value on the `category_code` field.
+
+<!-- 
+
+Filter: { category_code: null }
+
+
+ -->
+ 
+### 8. All the companies that have at least 100 employees but less than 1000. Retrieve only the `name` and `number of employees` fields.
+
+<!-- 
+
+Filter: { $and: [ {number_of_employees: { $gt: 99} }, {number_of_employees: { $lt: 1000} } ] }
+Project: {name: 1, number_of_employees: 1}
+Sort: {number_of_employees: 1}
+
+
+ -->
+
+### 9. Order all the companies by their IPO price in a descending order.
+
+<!-- 
+
+Sort: Sort: {"ipo.valuation_amount": -1}
+
+ -->
+
+### 10. Retrieve the 10 companies with most employees, order by the `number of employees`
+
+<!-- 
+
+Sort: {number_of_employees: -1}
+Limit: 10;
+
+ -->
+
+### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
+
+<!-- 
+
+Filter: {founded_month: { $gt: 5} }
+Limit: 1000
+
+ -->
+
+### 12. All the companies founded before 2000 that have an acquisition amount of more than 10.000.000
+
+<!-- 
+
+Filter: { $and: [ { founded_year: { $lt: 2000 } }, { "acquisitions.price_amount": { $gt: 10000000 } }, {"acquisition.price_amount": { $exists: true } }, {"acquisition.price_amount": { $ne: null } } ] }
```

Comment:
Bastaba con:

`{$and: [{founded_year: {$lt: 2000}}, {'acquisition.price_amount': {$gt: 10000000}}]}`